ccdproc/core.py

```diff
@@ -224,13 +228,12 @@ def ccd_process(ccd, oscan=None, trim=None, error=False, master_bias=None,
         raise TypeError('bad_pixel_mask is not None or numpy.ndarray.')
 
     # apply the gain correction
-    if isinstance(gain, Quantity):
-        nccd = gain_correct(nccd, gain)
-    elif gain is None:
-        pass
-    else:
+    if not (gain is None or isinstance(gain, Quantity)):
```

Comment:
the `isinstance(gain, Quantity)` is used 3 times in the function now. You could use an intermediate variable so it's only computed once.